### PR TITLE
chore(wdio): port form-associated spec from karma to wdio

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
This PR needs #5470 to land before it can be merged.

## What is the current behavior?

GitHub Issue Number: N/A

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: test/wdio/form-associated/cmp.test.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  describe('form associated custom element lifecycle callback', () => {
+    it('should trigger "formAssociated"', async () => {
+      const formEl = $('form');
+      expect(formEl).toHaveProperty('ariaLabel', 'formAssociated called');
+    });
+
+    it('should trigger "formResetCallback"', async () => {
+      const resetBtn = $('input[type="reset"]');
+      resetBtn.click();
```
Comment (christian-bromann) [inline test/wdio/form-associated/cmp.test.tsx:29]:
```suggestion
      await resetBtn.click();
```

File: test/wdio/form-associated/cmp.test.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+      await resetBtn.waitForStable();
+
+      const formEl = $('form');
+      expect(formEl).toHaveProperty('ariaLabel', 'formResetCallback called');
```
Comment (christian-bromann) [inline test/wdio/form-associated/cmp.test.tsx:34]:
```suggestion
      await expect(formEl).toHaveProperty('ariaLabel', 'formResetCallback called');
```

File: test/wdio/form-associated/cmp.test.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    });
+
+    it('should trigger "formDisabledCallback"', async () => {
+      const elm = document.body.querySelector('form-associated');
+      const formEl = $('form');
+
+      elm.setAttribute('disabled', 'disabled');
+
+      await formEl.waitForStable();
+      expect(formEl).toHaveProperty('ariaLabel', 'formDisabledCallback called with true');
```
Comment (christian-bromann) [inline test/wdio/form-associated/cmp.test.tsx:44]:
```suggestion
      await expect(formEl).toHaveProperty('ariaLabel', 'formDisabledCallback called with true');
```

File: test/wdio/form-associated/cmp.test.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+      elm.removeAttribute('disabled');
+      await formEl.waitForStable();
+      expect(formEl).toHaveProperty('ariaLabel', 'formDisabledCallback called with false');
```
Comment (christian-bromann) [inline test/wdio/form-associated/cmp.test.tsx:48]:
```suggestion
      await expect(formEl).toHaveProperty('ariaLabel', 'formDisabledCallback called with false');
```

File: test/wdio/form-associated/cmp.tsx [hunk 5]
```diff
@@ -5,8 +5,8 @@ import { AttachInternals, Component, h } from '@stencil/core';
   formAssociated: true,
 })
 export class FormAssociatedCmp {
-  @AttachInternals()
-  internals: ElementInternals;
+  // @ts-ignore typescript complains because this isn't initialized
```
Comment (christian-bromann) [inline test/wdio/form-associated/cmp.tsx:8]: Should we tweak the `tsconfig.json` for this?

Comment (alicewriteswrongs) [reply test/wdio/form-associated/cmp.tsx:8]: yeah I think that's a good idea, I turned `strictPropertyInitialization` off

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Review by tanner-reits — APPROVED** [review]

**Pull request merged**